src/timetable_cmd.cpp

```diff
@@ -519,16 +519,22 @@ void UpdateVehicleTimetable(Vehicle *v, bool travelling)
 	/* Before modifying waiting times, check whether we want to preserve bigger ones. */
 	if (!real_current_order->IsType(OT_CONDITIONAL) &&
 			(travelling || time_taken > real_current_order->GetWaitTime() || remeasure_wait_time)) {
-		/* Round up to the smallest unit of time commonly shown in the GUI (seconds) to avoid confusion.
+		/* Round up to the unit currently shown in the GUI for days and seconds.
+		 * Round up to seconds if currently used display style is ticks.
 		 * Players timetabling in Ticks can adjust later.
 		 * For trains/aircraft multiple movement cycles are done in one
 		 * tick. This makes it possible to leave the station and process
 		 * e.g. a depot order in the same tick, causing it to not fill
 		 * the timetable entry like is done for road vehicles/ships.
 		 * Thus always make sure at least one tick is used between the
 		 * processing of different orders when filling the timetable. */
-		uint time_to_set = CeilDiv(std::max(time_taken, 1), Ticks::TICKS_PER_SECOND) * Ticks::TICKS_PER_SECOND;
-
+		uint time_to_set;
+		if (_settings_client.gui.timetable_mode == TimetableMode::Days) {
+			time_to_set = CeilDiv(std::max(time_taken, 1), Ticks::DAY_TICKS) * Ticks::DAY_TICKS;
+		} else {
+			time_to_set = CeilDiv(std::max(time_taken, 1), Ticks::TICKS_PER_SECOND) * Ticks::TICKS_PER_SECOND;
+		}
```

Comment:
Feels slightly clunky to me. Maybe:

```suggestion
		uint factor = _settings_client.gui.timetable_mode == TimetableMode::Days ? Ticks::DAY_TICKS : Ticks::TICKS_PER_SECOND;
		uint time_to_set = CeilDiv(std::max(time_taken, 1), factor) * factor;
```

?